### ISSUE TITLE
Improve the test coverage

### ISSUE THREAD

**Issue by forgedhallpass** [OWASP/www-project-csrfguard#24]
Unit and integration tests should be written to make sure that the functionalities still work as expected. It would limit the possible regression issues caused by new development/refactorings, especially since there are several possible configuration permutations.

Detailed manual testing scenarios, executions flows would be very helpful as well because it could also act as a documentation/functional requirements.

**Comment by forgedhallpass** [comment]
Related commits:
* https://github.com/OWASP/www-project-csrfguard/commit/23e586ab609aa7a5a484136f986cbef26c5dfb0e
* https://github.com/OWASP/www-project-csrfguard/commit/1c3bb4635aed2f5a1d2f2c561af88e05df53cf8c
* https://github.com/OWASP/www-project-csrfguard/commit/18ab5cc4ebaa0bace0f6c2c4b71ecc494ca1b3aa
* https://github.com/OWASP/www-project-csrfguard/commit/c787303d4613247f0dc2b8df875d867e80a95ca7

**Comment by yashpandey06** [comment]
@forgedhallpass , Hey there , I am picking up this issue to work !...thanks !

Also can you please share the communication channels where i can ask doubts ? it would be helpful .

**Comment by forgedhallpass** [comment]
Hello @yashpandey06,

Thank you for your interest in contributing to the project.
You can ask your questions here, in the OWASP Slack channel or via e-mail.

**Comment by yashpandey06** [comment]
@forgedhallpass yeah , thanks for your kind response, just confirming as I can see this issue was opened a long time back , do we still have this issue on priority list ?

**Comment by forgedhallpass** [comment]
@yashpandey06 we have two issues created. One to write integration tests and this one to improve the test coverage either by writing more unit/integration tests or by identifying/implementing manual test use cases that can be implemented in the bundled test application. These are ongoing efforts and contributions are always welcome.
